### PR TITLE
clean up ffi from pyframe to pyhash

### PR DESCRIPTION
**Pull request by deantvv**
Try to cleanup ffi module. Related to #1289.

### CODE REVIEW

File: src/ffi/pyframe.rs [hunk 1]
```diff
@@ -8,5 +8,4 @@ opaque_struct!(PyFrameObject);
 extern "C" {
     pub fn PyFrame_GetLineNumber(f: *mut PyFrameObject) -> c_int;
 }
-// skipped PyFrame_GetLineNumber
```
Comment (davidhewitt) [inline src/ffi/pyframe.rs:11]: This symbol is still present in the C header? 

https://github.com/python/cpython/blob/a40675c659cd8c0699f85ee9ac31660f93f8c2f5/Include/pyframe.h#L15

File: src/ffi/pyhash.rs [hunk 2]
```diff
@@ -1,6 +1,18 @@
 use crate::ffi::pyport::{Py_hash_t, Py_ssize_t};
-use std::os::raw::{c_char, c_int, c_void};
+use std::os::raw::{c_char, c_int, c_ulong, c_void};
 
+// skipped non-limited _Py_HashDouble
+// skipped non-limited _Py_HashPointer
+// skipped non-limited _Py_HashPointerRaw
+// skipped non-limited _Py_HashBytes
```
Comment (davidhewitt) [inline src/ffi/pyhash.rs:7]: Looks like we do define this down below, perhaps move it up here?

**Review by davidhewitt — APPROVED** [review]
👍 thanks for doing these. Just two comments, once those are resolved let's merge!

File: src/ffi/pyframe.rs (hunk 1 above)
Comment (deantvv) [reply src/ffi/pyframe.rs:11]: Maybe I get it wrong, but isn't there a `PyFrame_GetLineNumber` at line 9?

File: src/ffi/pyhash.rs (hunk 2 above)
Comment (deantvv) [reply src/ffi/pyhash.rs:7]: Yeah, I missed this!

File: src/ffi/pyframe.rs (hunk 1 above)
Comment (davidhewitt) [reply src/ffi/pyframe.rs:11]: Haha yes sorry that's my bad for reviewing at midnight 😮‍💨

**Review by davidhewitt — APPROVED** [review]
👍 as well as relocating `_Py_HashBytes` it looks like `PyHash_GetFuncDef` should also be `#[cfg(not(any(Py_LIMITED_API, PyPy)))]`

With those tweaked and CI green we can merge :)

**Pull request merged**